### DISCUSSION TITLE
Poco C++ On Android armv7a not listing network interface list

### DISCUSSION THREAD

**Discussion by Hakai-Shin** [pocoproject/poco#4386] [Q&A]
I have built poco for android armv7a and using under ndk code. I have written code to list all the available interfaces in and android armv7a based device. The build is successful and I have added logs. when running on the device, android logcat is not showing any logs beyond the function call to get network interface list.

`LOGV("open");

Poco::Net::NetworkInterface::AddressList interfaces = Poco::Net::NetworkInterface::addressList();

LOGV("got interface");
` 

I am getting only "open" log line but no "got interface". There is not crash logs in adb logcat.

**Comment by Hakai-Shin** [comment]
I have added try catch block around the code, and not it shows "Not implemented: Not implemented in Android", is there not support for multicast socket in poco android

**Comment by Bjoe** [comment]
@Hakai-Shin I'm little bit confused. The code snippit shows how you get the IP addresses from an Interface but not all Interfaces. I think you mean this code snippit:
`auto netList = Poco::Net::NetworkInterface::list();`

Can you also add the info how you build the Poco library for android and for which API level ?.
Did you also set the "network permission" on the APK ?
